esmvalcore/_recipe_checks.py

```diff
@@ -220,16 +219,88 @@ def extract_shape(settings):
                 "{}".format(', '.join(f"'{k}'".lower() for k in valid[key])))
 
 
-def valid_multimodel_statistic(statistic):
-    """Check that `statistic` is a valid argument for multimodel stats."""
-    valid_names = ['std'] + list(STATISTIC_MAPPING.keys())
+def _verify_statistics(statistics, step):
+    """Raise error if multi-model statistics cannot be verified."""
+    valid_names = ["mean", "median", "std", "min", "max"]
     valid_patterns = [r"^(p\d{1,2})(\.\d*)?$"]
-    if not (statistic in valid_names
-            or re.match(r'|'.join(valid_patterns), statistic)):
+
+    for statistic in statistics:
+        if not (statistic in valid_names
+                or re.match(r'|'.join(valid_patterns), statistic)):
+            raise RecipeError(
+                "Invalid value encountered for `statistic` in preprocessor "
+                f"{step}. Valid values are {valid_names} "
+                f"or patterns matching {valid_patterns}. Got '{statistic}'.")
+
+
+def _verify_span_value(span):
+    """Raise error if span argument cannot be verified."""
+    valid_names = ('overlap', 'full')
+    if span not in valid_names:
+        raise RecipeError(
+            "Invalid value encountered for `span` in preprocessor "
+            f"`multi_model_statistics`. Valid values are {valid_names}."
+            f"Got {span}.")
+
+
+def _verify_groupby(groupby):
+    """Raise error if groupby arguments cannot be verified."""
+    if not isinstance(groupby, list):
+        raise RecipeError(
+            "Invalid value encountered for `groupby` in preprocessor "
+            "`multi_model_statistics`.`groupby` must be defined as a "
+            f"list. Got {groupby}.")
+
+
+def _verify_keep_input_datasets(keep_input_datasets):
+    if not isinstance(keep_input_datasets, bool):
         raise RecipeError(
-            "Invalid value encountered for `statistic` in preprocessor "
-            f"`multi_model_statistics`. Valid values are {valid_names} "
-            f"or patterns matching {valid_patterns}. Got '{statistic}.'")
+            "Invalid value encountered for `keep_input_datasets`."
+            f"Must be defined as a boolean. Got {keep_input_datasets}."
+        )
+
+
+def _verify_arguments(given, expected):
+    """Raise error if arguments cannot be verified."""
+    for key in given:
+        if key not in expected:
+            raise RecipeError(
+                f"Unexpected keyword argument encountered: {key}. Valid "
+                f"keywords are: {expected}.")
+
+
+def multimodel_statistics(settings):
```

Comment:
I would use a different function name, this is the same as the actual preprocessor, also maybe make it private too since it may get picked up in the API documentation

Reply:
I'll change the name but plenty of other checks like this one are not private (`valid_time_selection`, `reference_for_bias_preproc`) so I am not sure why this and the ensemble one should be any different.

Reply:
ah right, I'd have made them all private, prob legacy of ye olde times, yes please change the name, we need to be careful not to reuse the same function names in different modules